common/src/main/java/io/homeassistant/companion/android/common/sensors/DisplaySensorManager.kt

```diff
@@ -98,4 +159,110 @@ class DisplaySensorManager : SensorManager {
             mapOf()
         )
     }
+
+    private fun updateScreenOrientation(context: Context) {
+        if (!isEnabled(context, screenOrientation)) {
+            return
+        }
+
+        @Suppress("DEPRECATION")

```

Comment:
I hadn't looked at the docs yet, but now I did and it [literally says](https://developer.android.com/reference/kotlin/android/view/Display.html?hl=en#getorientation): "Deprecated: use getRotation" since forever (API 15). Is it really useful to have both then? Yes they are technically different but orientation seems mostly like a convenient way to know that 0 = portrait/landscape which the user will almost certainly already know.

Reply:
no this deprecation is for the square display type lol https://developer.android.com/reference/android/content/res/Configuration.html#ORIENTATION_SQUARE

I had removed teh rotation one already in a previous commit :)

the actual API used is not deprecated :)
https://developer.android.com/reference/android/content/res/Configuration.html#orientation

> Is it really useful to have both then?

its to help differentiate between reverse landscape/portrait which are not real states but still a different orientation, the original request wanted to know if the device is horizontal and this should help with that and additional states.

Reply:
Oops my bad for mixing it up, sorry.

> its to help differentiate between reverse landscape/portrait which are not real states but still a different orientation, the original request wanted to know if the device is horizontal and this should help with that and additional states.

So a convenience, but not for the original request :)

Horizontal is not a state we provide so you still have to determine if that is landscape/portrait/0/90/180/270 yourself. The only state you cannot determine from rotation is 'square' but the docs say that isn't used anymore on APIs the app supports.

Reply:
yea the angles are device dependent so they will vary from tablet to phone in this case so I figured best to leave it in their hands to determine what to do. The square one is weird but doesnt surprise me if a device like NS panel pro has it lol